data/chests.py

```diff
@@ -105,6 +106,27 @@ def shuffle_random(self):
             elif chest.type == Chest.ITEM:
                 chest.contents = self.items.get_random()
 
+        if self.args.no_trash_chests:
```

Comment:
This change only applies if with a -ccsr # (where number is non-zero). I recommend making it apply for all other chest options as well by moving it into its own method called by mod.